### ISSUE TITLE
Collaboration between Dask and Modin Dataframe

### ISSUE THREAD

**Issue by devin-petersohn** [dask/dask#7266]
Hi folks, I'm Devin from the Modin group.

Between our groups, I believe there's a high potential for collaboration and improving the data science experience for Dask and Modin users. Due to the complexity/differences in the architectures of Dask and Modin Dataframes, there's a lot to consider. I'll outline a proposal for next steps here.

---

**Step 1: Interoperability between Modin and other Dask libraries** 
* Conversion of Modin Dataframe to:
  * Dask Dataframe
  * Dask Array
  * Other useful structures? There's a power law with Dask's other structures so it probably makes sense to prioritize based on commonality
* Interoperability between Modin Dataframe and Dask XGBoost

**Step 2: Learn from each other**
* Are there issues that are unable to be compromised (e.g. must sort row label values, must support certain operators, etc.)?
* What do Dask/Modin Dataframe users care about?
* What does Dask/Modin Dataframe allow that Modin/Dask does not?
* Is there enough overlap in our goals to combine forces?

**Step 3: Combining forces**
* How to do this will be based on Step 2 outcomes

Steps 1 and 2 can be worked on simultaneously. Regardless of the Step 2 outcomes I think Step 1 is a no-brainer.

---

Here are my high level thoughts:
* I would like Modin's users to be able to use other Dask objects and utilities
* I think Dask Dataframe users would benefit from the academic research work we've done (some yet unpublished)
* I also think there are engineering features that Modin has that would interest Dask Dataframe users as well

**Comment by devin-petersohn** [comment]
>Regarding dask-xgboost it seems like the easiest solution would be to make it easy to convert a Modin dataframe into a Dask dataframe and then use the existing solution. My guess is that this is relatively straightforward.

Yes, I agree it would likely be straightforward. There are other options, like adding the ability to create a `DaskDMatrix` from a Modin Dataframe. I'd personally be in favor of the latter, but those changes would likely go in xgboost itself.

>Regarding conversion, my understanding is that you care about cases where there are very many columns.

Not completely. Modin is designed to scale horizontally, but there is more to the design than handling a large number of columns. Dask Dataframe can handle hundreds of columns, and we don't see many users who have much more than that (maybe a handful have in the low 1000s). Conversion to/from Dask Dataframe and Modin would benefit everyone. I am familiar with Xarray, I point people there from time to time, but I believe we're solving different problems.

>I don't think that there are cohesive answers to these questions.

I'm sure there are concrete answers to the first and last question. The second question is perhaps too broad, and maybe could be phrased as "Are there any features that any Dask/Modin users could not operate without?".  I do think the questions are important to consider, not as pitting Modin vs Dask rather in the interest of learning from each other. Comparisons are hard, but there's some common ground and it's likely a good place to start if we want to "combine forces" (do we?). It'd also be good to get acquainted with each other's way of thinking.

>I would love to hear more about these.

I think there's a cost/benefit to be considered here, so I'd first like to list a couple of the lowest hanging fruits 😄. These are also mutually beneficial.

1. Conversion to and from Modin from Dask Dataframe
  a. Should be easy for no-copy partition conversion
  b. Modin can handle the partitions Dask gives it and provide the partitions in a layout Dask can directly consume
  c. (Downside) Metadata might be hard (e.g. unsorted row labels)
2. Add ability to query Dask dataframe directly using Modin
  a. Benefit of Modin's architecture without directly changing Dask
  b. (Downside) Not everything will be supported in Dask dataframe that Modin on Dask currently supports

I think it's easier to justify the non-trivial time spent on building these features if there's some common middle-long term goal in mind for collaborating. Expectations should be set first, I believe.

**Comment by YarShev** [comment]
Hi folks, it's been a while since opening this issue :) 3 years later Modin added support for conversion of a Modin DF to a Dask DF and vice versa. By converting a Modin DF to a Dask DF, there will be opportunity for conversion to other Dask data formats, as well as for usage of Dask XGBoost. We are going to open a PR to update the Dask documentation so after that one can consider this issue as resolved.
